ingestion-edge/ingestion_edge/disk_queue.py

```diff
@@ -0,0 +1,70 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+
+from .conf import QUEUE_FILE
+from .conf import QUEUE_POSITION_FILE
+from flask import Blueprint
+from google.cloud.pubsub import PublisherClient
+from os import remove
+from os.path import exists
+from threading import Lock
+from typing import Tuple
+import ujson
+
+disk_queue = Blueprint("disk_queue", __name__)
+client = PublisherClient()
+read_lock = Lock()
+write_lock = Lock()
+
+
+def write(**kwargs):
+    with write_lock:
+        # TODO disable autoscaling
+        with open(QUEUE_FILE, "a") as q:
+            q.write(ujson.dumps(kwargs) + "\n")
+            q.flush()
+
+
+@disk_queue.route("/__flush__")
+def flush_queue() -> Tuple[str, int]:
+    if not exists(QUEUE_FILE):
+        # early return for empty queue
+        return "", 204
```

Comment:
"if exists, open" is generally frowned upon, as it can introduce race conditions. This also doesn't guard again a queue file that is unreadable for other reasons (such as permissions). The standard advice is to open the file and catch the error, reacting appropriately.